packages/relayer/indexer/config.go

```diff
@@ -70,6 +71,14 @@ func NewConfigFromCliContext(c *cli.Context) (*Config, error) {
 		SyncMode:                            SyncMode(c.String(flags.SyncMode.Name)),
 		ETHClientTimeout:                    c.Uint64(flags.ETHClientTimeout.Name),
 		NumLatestBlocksToIgnoreWhenCrawling: c.Uint64(flags.NumLatestBlocksToIgnoreWhenCrawling.Name),
+		TargetBlock: func() *uint64 {
+			if c.IsSet(flags.TargetBlock.Name) {
+				value := c.Uint64(flags.TargetBlock.Name)
+				return &value
+			} else {
```

Comment:
we don't need this else, since the if statement returns. it can just be return nil.

actually this doesn't need to be a function at all. IMO, if you can just do this check above the struct?

Reply:
Removed the else. 

It's to check if `TARGET_BLOCK_NUMBER` is set. If it's not, then `TargetBlockNumber` is set to nil. 

`c.Uint64(flags.TargetBlock.Name)` will return 0 if `TARGET_BLOCK_NUMBER` is not set, which doesn't work.

Reply:
you can use if c.IsSet above during the config creation though, and still set it to nil. this doesn't need to be a function. but if you want it to be a function then okay